PhysicsTools/Heppy/python/analyzers/objects/JetAnalyzer.py

```diff
@@ -83,13 +105,19 @@ def __init__(self, cfg_ana, cfg_comp, looperName):
           if   doResidual == "MC":   doResidual = self.cfg_comp.isMC
           elif doResidual == "Data": doResidual = not self.cfg_comp.isMC
           elif doResidual not in [True,False]: raise RuntimeError, "If specified, applyL2L3Residual must be any of { True, False, 'MC', 'Data'(default)}"
-          GT = getattr(cfg_comp, 'jecGT', mcGT if self.cfg_comp.isMC else dataGT)
+          GTs = getattr(cfg_comp, 'jecGT', mcGT if self.cfg_comp.isMC else dataGT)
           # Now take care of the optional arguments
           kwargs = { 'calculateSeparateCorrections':calculateSeparateCorrections,
                      'calculateType1METCorrection' :calculateType1METCorrection, }
           if kwargs['calculateType1METCorrection']: kwargs['type1METParams'] = cfg_ana.type1METParams
           # instantiate the jet re-calibrator
-          self.jetReCalibrator = JetReCalibrator(GT, cfg_ana.recalibrationType, doResidual, cfg_ana.jecPath, **kwargs)
+          self.jetReCalibrators={}
+          self.runsGT=[]
+          for iGT in range(0,len(GTs)):
+              self.jetReCalibrators[iGT] = JetReCalibrator(GTs[iGT][1], cfg_ana.recalibrationType, doResidual, cfg_ana.jecPath, **kwargs)
+              self.runsGT.append(GTs[iGT][0])
+          self.runsGT.append(1000000000)
```

Comment:
some code cleanup:
 * make self.jetReCalibrators a list instead of a map, and append
 * you don't need the `self.runsGT.append(1000000000)` since `bisect` will return work even without it
 * you could do `for (run,GT) in GTs:` instead of iterating on the index, but that's not a big deal.